src/shared/helpers.h

```diff
@@ -165,4 +165,16 @@ RC_UNUSED RC_PRINTF(2,3) static int xasprintf(char **strp, const char *fmt, ...)
 	return len;
 }
 
+RC_UNUSED static ssize_t xgetline(char **restrict lineptr, size_t *restrict n, FILE *restrict stream)
+{
+	ssize_t ret = getline(lineptr, n, stream);
+	if (ret == -1)
+		return -1;
+
+	if ((*lineptr)[ret - 1] == '\n')
```

Comment:
an empty file will return -1 with feof set, and non empty has the minimum being 1 since \n, so i can't think of when ret would be 0, and the manpage doesn't mention so

but for safety, i can do instead:

```
if (ret <= 0)
    return ret;
```